public_html/modules.php

```diff
@@ -0,0 +1,222 @@
+                                    // catch incorrectly formatted yamls
+                                    if (isset($tool['documentation'])) {
+                                        $tmp_tool = [];
+                                        $tmp_tool[key($tool)] = array_slice($tool, 1);
+                                        $tool = $tmp_tool;
+                                    }
+                                    foreach ($tool as $name => $tool_value) {
+                                        // don't print tools if it has the same name (and therefore usually same description) as the module
+                                        if ($module['name'] != $name) {
+
+                                            $tool_text .= '<div>';
+                                            $tool_text .= '<span>' . $name . ': </span>';
+                                            $description = $tool_value['description'];
+                                            $tool_text .= '<span class="text-small description ' . $module['name'] . '-description" >' .  $description . '</span>';
+                                            $tool_text .= '</div>';
+                                        }
+                                        $tool_text .= '</ul>';
+                                    }
+                                }
+                                if (substr_count($tool_text, '<div>') > 0) {
+                                    $tool_text = '<label class="text-success">Tools</label>' . $tool_text;
+                                }
+                                echo $tool_text;
+                                ?>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+        </div>
+    <?php endforeach; ?>
+    </div>
+    <nav aria-label="Module page navigation ">
+        <ul class="pagination">
+            <?php
+            $disable = $current_page - 1 == 0 ? 'disabled' : '';
+            $params = add_update_url_param('page', $current_page - 1);
+            echo '<li class="page-item ' . $disable . '"><a class="page-link" href= "/modules?' . $params . '">Previous</a></li>';
+            for ($page_number = 1; $page_number <= $total_pages; $page_number++) {
+                $params = add_update_url_param('page', $page_number);
+                $active = $current_page == $page_number ? 'active' : '';
+                echo '<li class="page-item ' . $active . '"><a class="page-link" href= "/modules?' . $params . '">' . $page_number . ' </a></li>';
+            }
+            $params = add_update_url_param('page', $current_page + 1);
+            $disable = $current_page - $total_pages == 0 ? 'disabled' : '';
+            echo '<li class="page-item ' . $disable . '"><a class="page-link" href= "/modules?' . $params . '">Next</a></li>';
+            ?>
+        </ul>
+    </nav>
+</div>
+
+<?php include('../includes/footer.php'); ?>
```

Comment:
Habit brought from my WordPress plugin days, but safer to just not close the PHP tags in a file if you don't need to..

```suggestion
<?php include('../includes/footer.php');

```